controller_manager/test/test_controllers_chaining_with_controller_manager.cpp

```diff
@@ -975,10 +977,91 @@ TEST_P(
     lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, diff_drive_controller->get_state().id());
 }
 
-// TODO(destogl): Add test case with controllers added in "random" order
-//
 // new value: "START_DOWNSTREAM_CTRLS" --> start "downstream" controllers in a controllers chain
 //
+TEST_P(TestControllerChainingWithControllerManager, test_chained_controllers_random_order)
```

Comment:
I'm sorry but I don't get what this test does. It feels like this is testing multiple components and various random things... We should clarify the implementation of this. I don't get why the internal counter check should be different, I don't get why a "random order test" starts with "add all controllers IN RESERVE EXECUTION ORDER" ?? so it's ordered, not random :D